src/opensynth/models/faraday/model.py

```diff
@@ -372,26 +477,29 @@ def sample_gmm(
               Decoder output (KWH), month label, dow label
         """
         gmm_samples = self.gmm.sample(n_samples)[0]
+
+        # Parse labels and profiles
         gmm_kwh = gmm_samples[:, : self.vae_module.latent_dim]
-        gmm_mth = np.round(gmm_samples[:, -2], decimals=0).astype(int)
-        gmm_dow = np.round(gmm_samples[:, -1], decimals=0).astype(int)
+        gmm_labels: dict[str, torch.tensor] = {}
+        # Abstract the labels and round numerical values to integers
+        for i, feature in enumerate(self.vae_module.feature_list):
+            index = self.get_index(self.vae_module.feature_list, i)
+            gmm_labels[feature] = np.round(
+                gmm_samples[:, index], decimals=0
+            ).astype(int)
```

Comment:
Should all features be converted to `int` types? Could it be possible that the user intended other input features to be of a different datatype? Or is `int` a requirement of features?

Reply:
Theoretically no, GMM by default handle floats, but it's a bit cumbersome to have to cater for non-integer encoded features. I'll add this to docstrings and end users can always subclass `FaradayModel` and overwrite this method to support float labels if they so wish.